src/solver/solver.cxx

```diff
@@ -850,17 +829,17 @@ void Solver::loop_vars_op(int jx, int jy, BoutReal *udata, int &p, SOLVER_VAR_OP
     for(const auto& f : f2d) {
       if(bndry && !f.evolve_bndry)
         continue;
-      (*f.var)(jx, jy) = udata[p];
+      (*f.var)[i2d] = udata[p];
       p++;
     }
     
-    for (jz=0; jz < mesh->LocalNz; jz++) {
+    for (int jz=0; jz < nz; jz++) {
       
       // Loop over 3D variables
       for(const auto& f : f3d) {
         if(bndry && !f.evolve_bndry)
           continue;
-        (*f.var)(jx, jy, jz) = udata[p];
+        (*f.var)[Ind3D(i2d.ind*nz + jz)] = udata[p];
```

Comment:
I've finally got around to updating this, but it ends up looking like:

```cpp
        (*f.var)[f.var->getMesh()->ind2Dto3D(i2d, jz)] = udata[p];
```

Given that we're already looping over `mesh->LocalNz`, I'm not 100% sure this is any better. I suspect that `solver` probably assumes all the fields it's evolving are on the same `mesh` anyway...